x-pack/plugins/ml/public/application/explorer/anomalies_map.tsx

```diff
@@ -0,0 +1,268 @@
+              name: 'count',
+              origin: FIELD_ORIGIN.JOIN,
+            },
+            useCustomColorRamp: false,
+          },
+        },
+        lineColor: {
+          type: STYLE_TYPE.DYNAMIC,
+          options: { fieldMetaOptions: { isEnabled: true } },
+        },
+        lineWidth: { type: STYLE_TYPE.STATIC, options: { size: 1 } },
+      },
+      isTimeAware: true,
+    },
+    visible,
+    type: 'VECTOR',
+  };
+};
+
+interface Props {
+  anomalies: AnomaliesTableRecord[];
+  jobIds: string[];
+}
+
+interface MLEMSTermJoinConfig extends EMSTermJoinConfig {
+  jobId: string;
+}
+
+export const AnomaliesMap: FC<Props> = ({ anomalies, jobIds }) => {
+  const [EMSSuggestions, setEMSSuggestions] = useState<
+    Array<MLEMSTermJoinConfig | null> | undefined
+  >();
+  const {
+    services: { maps: mapsPlugin },
+  } = useMlKibana();
+
+  const getEMSTermSuggestions = useCallback(async (): Promise<void> => {
+    if (!mapsPlugin) return;
+
+    const suggestions = await Promise.all(
+      jobIds.map(async (jobId) => {
+        const entityValues = new Set<string>();
+        let entityName;
+        for (let i = 0; i < anomalies.length; i++) {
+          if (
+            jobId === anomalies[i].jobId &&
+            anomalies[i].entityValue !== '' &&
+            anomalies[i].entityValue !== undefined &&
+            anomalies[i].entityName !== '' &&
+            anomalies[i].entityName !== undefined
+          ) {
+            entityValues.add(anomalies[i].entityValue);
+
+            if (!entityName) {
+              entityName = anomalies[i].entityName;
+            }
+          }
+
+          if (
+            // convert to set so it's unique values
+            entityValues.size === MAX_ENTITY_VALUES
+          )
+            break;
+        }
+
+        const suggestion: EMSTermJoinConfig | null = await mapsPlugin.suggestEMSTermJoinConfig({
+          emsLayerIds: COMMON_EMS_LAYER_IDS,
+          sampleValues: Array.from(entityValues),
+          sampleValuesColumnName: entityName || '',
+        });
+        if (suggestion) {
+          return { jobId, ...suggestion };
+        }
+        return suggestion;
+      })
+    );
+
+    setEMSSuggestions(suggestions.filter((s) => s !== null));
+  }, [...jobIds]);
+
+  useEffect(
+    function getInitialEMSTermSuggestions() {
+      if (anomalies && anomalies.length > 0) {
+        getEMSTermSuggestions();
+      }
+    },
+    [...jobIds]
+  );
+
+  const layerList = useMemo(() => {
+    let layers: VectorLayerDescriptor[] = [];
+    // Loop through suggestions list and make a layer for each
+    if (EMSSuggestions?.length) {
+      let count = 0;
+      layers = EMSSuggestions.reduce(function (result, suggestion) {
+        if (suggestion) {
+          const visible = count === 0;
+          result.push(getChoroplethAnomaliesLayer(anomalies, suggestion, visible));
+          count++;
+        }
+        return result;
+      }, [] as VectorLayerDescriptor[]);
```

Comment:
When you call `setEMSSuggestions` (line 198), you filter out `null` values. Do you need this extra check here? Seems like this code can be simplified by using `map` instead of `reduce` with an external counter. 

```suggestion
      layers = EMSSuggestions.map((suggestion, i) => {
          const visible = i === 0;
          return getChoroplethAnomaliesLayer(anomalies, suggestion, visible);
      });
```

In case you still need to filter out empty suggestions, you can put `.filter` in front of the `.map`